XenAdmin/Wizards/RollingUpgradeWizard/RollingUpgradeWizardPrecheckPage.cs

```diff
@@ -294,6 +297,27 @@ where check.CanRun()
             if (deprecatedSRsChecks.Count > 0)
                 groups.Add(new CheckGroup(Messages.CHECKING_DEPRECATED_SRS, deprecatedSRsChecks));
 
+
+            // EUA check
+            if (InstallMethodConfig != null && InstallMethodConfig.TryGetValue("url", out var uriText))
+            {
+                Uri targetUri = null;
+                try
+                {
+                    targetUri = new Uri(uriText);
+                }
+                catch (Exception ex)
+                {
+                    Log.Error(ex);
+                }
+
+                var euaCheck = GetPermanentCheck("EUA", new UpgradeRequiresEua(hostsToUpgrade, targetUri));

```

Comment:
Not sure I agree with this. Moving the filtering to the caller means that whenever `UpgradeRequiresEua` is called, the developer needs to know and remember to filter the list before passing it to it. I think it's best if the `Check` itself knows that it only applies to post82x hosts. NowI know we will probably only ever have this `Check` called within this method, but it's just safer this way. Thoughts?

Reply:
Moving the filtering to the caller may be needed in order to hide the check completely from the user if their system is not relevant. For example, running a check on havana for a feature that was added on naples will not cause an issue since the host will be filtered out, but the anachronism may confuse the user. At least that was the thinking - I'm not sure we get it right in all cases, the checks may need audit at some point (possibly needed by the merge epic). I suppose checking for EUA may not belong in this category. Ok, leave it for now, if needed we can revisit it later if auditing other checks.